### PR TITLE
[fix](MTMV) Support refreshing data manually

### LINKED ISSUE #20105: [Bug] When refreshing the mv, prompt no job
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/doris/issues?q=is%3Aissue) and found no similar issues.

### Version

master

### What's Wrong?

CREATE MATERIALIZED VIEW mv2 BUILD DEFERRED REFRESH COMPLETE ON DEMAND DISTRIBUTED BY HASH(k1) 
BUCKETS 6 
PROPERTIES (
    "replication_num" = "1"
)
as select user1.k1,user1.k3,user2.k4 from user1 join user2 on user1.k1=user2.k1; 

REFRESH MATERIALIZED VIEW mv2;

ERROR 1105 (HY000): errCode = 2, detailMessage = No job find for the MaterializedView default_cluster:zd.mv2 .

### What You Expected?

normal

### How to Reproduce?

_No response_

### Anything Else?

_No response_

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by zddr**
# Proposed changes

Issue Number: close #20105 

## Problem summary

Describe your changes.

Add a manual task to all materialized view

## Checklist(Required)

* [ ] Does it affect the original behavior
* [ ] Has unit tests been added
* [ ] Has document been added or modified
* [ ] Does it need to update dependencies
* [ ] Is this PR support rollback (If NO, please explain WHY)

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by zddr** [comment]
run buildall

File: fe/fe-core/src/main/java/org/apache/doris/mtmv/MTMVJobFactory.java [hunk 1]
```diff
@@ -70,6 +59,7 @@ public static List<MTMVJob> buildJob(MaterializedView materializedView, String d
         if (!isRunPeriodJobImmediate && materializedView.getBuildMode() == BuildMode.IMMEDIATE) {
             jobs.add(genOnceJob(materializedView, dbName));
         }
```
Comment (SaintBacchus) [inline fe/fe-core/src/main/java/org/apache/doris/mtmv/MTMVJobFactory.java:61]: 为什么这里要每次加一个任务啊, 这个`genManualJob`任务也应该是立即执行的吧

File: fe/fe-core/src/main/java/org/apache/doris/mtmv/MTMVJobFactory.java [hunk 2]
```diff
@@ -42,18 +41,8 @@ public class MTMVJobFactory {
...
         //can not generate a job when creating a temp materialized view.
-        if (materializedView.getName().startsWith(FeConstants.TEMP_MATERIZLIZE_DVIEW_PREFIX)) {
-            return false;
-        }
-        if (buildMode == BuildMode.IMMEDIATE) {
-            return completeRefresh;
-        } else {
-            return completeRefresh && triggerInfo != null && triggerInfo.getRefreshTrigger() == RefreshTrigger.INTERVAL;
-        }
+        return !materializedView.getName().startsWith(FeConstants.TEMP_MATERIZLIZE_DVIEW_PREFIX);
```
Comment (SaintBacchus) [inline fe/fe-core/src/main/java/org/apache/doris/mtmv/MTMVJobFactory.java:45]: 如果创建物化视图里面, 没有定义任务, 这里也要产生吗

**Comment by zddr** [comment]
run buildall

**Comment by zddr** [comment]
run buildall

**Comment by SaintBacchus** [comment]
LGTM

**Comment by zddr** [comment]
run buildall

**Comment by zddr** [comment]
run buildall

**Comment by zddr** [comment]
run buildall

**Comment by zddr** [comment]
run buildall

**Comment by zddr** [comment]
run p0

**Comment by zddr** [comment]
run buildall

**Comment by adonis0147** [comment]
run buildall

**Review by adonis0147 — DISMISSED** [review]
LGTM

**Comment by SaintBacchus** [comment]
LGTM

**Comment by zddr** [comment]
run p0

**Comment by zddr** [comment]
run buildall

**Comment by zddr** [comment]
run p0

**Comment by zddr** [comment]
run buildall

**Comment by zddr** [comment]
run buildall

**Review by morningman — APPROVED** [review]
LGTM

**Pull request merged**